node/src/actors/chain_manager/mod.rs

```diff
@@ -956,15 +958,19 @@ impl ChainManager {
             self.current_epoch,
             self.epoch_constants,
         ) {
-            if let Transaction::Commit(_commit) = &msg.transaction {
-                let timestamp_mining = epoch_constants
-                    .block_mining_timestamp(current_epoch)
-                    .unwrap();
-
-                if timestamp_now > timestamp_mining {
-                    let e = ChainManagerError::TooLateToCommit;
-                    log::debug!("{}", e);
-                    return Box::new(actix::fut::err(e.into()));
+            let timestamp_mining = epoch_constants
+                .block_mining_timestamp(current_epoch)
+                .unwrap();
+
+            if timestamp_now > timestamp_mining {
```

Comment:
Race condition: this can happen before the mining starts, is that a problem?

Reply:
not really, worst case a commit that arrive in the limit of block creation that could be included is not